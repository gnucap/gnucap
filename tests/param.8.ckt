'parameter test
v1 1 0 dc a ac b
r1 1 2 c
r2 2 0 d
.param a=1 b=2 c=3
.param
.print op v(nodes) iter(0)
.op
.status notime
.end
