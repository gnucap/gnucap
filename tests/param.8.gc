verilog

module main(\1 ,\2 );
  parameter real v;
  parameter integer test = 2;
  vsource #(.dc(v + test / 3)) v1(\2 ,\1 )
endmodule

\\ ground \0

main #(.v(1)) m1(\0 ,\1 )
main #(.v(2)) m2(\0 ,\2 )
main #(.v(a)) m2(\0 ,\3 )

print dc v(nodes)
param a=1
dc
param a=2
dc
param a=3
dc
