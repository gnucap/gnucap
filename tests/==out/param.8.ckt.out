'parameter test 
 a= 1. b= 2. c= 3.
parameter d not specified, using default
parameter c+d not specified, using default
#           v(1)       v(2)       v(3)       iter(0)    r(r3)     
r3: short circuit
parameter d not specified, using default
parameter c+d not specified, using default
r2: short circuit
r3: short circuit
r2: short circuit
r3: short circuit
r2: short circuit
r3: short circuit
r2: short circuit
r3: short circuit
 27.        1.         3.3333u    0.         4.         10.u      
Gnucap   System status
iterations: op=4, dc=0, tran=0, fourier=0, total=4
transient timesteps: accepted=0, rejected=0, total=0
nodes: user=3, subckt=0, model=0, total=3
dctran density=55.6%, ac density=55.6%
